### PR TITLE
SEN-2275:

### PR DESCRIPTION
**Pull request by LauraVerheyde**
* added 2 recipes about @Scope annotation on singleton classes

### CODE REVIEW

**Comment by LauraVerheyde** [comment]
More info in [SEN-2275](https://securecodewarrior.atlassian.net/browse/SEN-2275)

File: recipes/Java/Spring/Boot/descriptions/Code_quality__Do_not_add__Scope__singleton___on_a_Spring__rest_controller__service_or_repository.html [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+}</pre>
+  <br>
+  <b>After</b>
+<pre>@Controller
+public class UserController {
+}</pre>
+  <br>
+  <b>Resources</b>
+<ul>
+  <li><a href="https://docs.spring.io/spring-framework/docs/3.0.0.M3/reference/html/ch04s04.html">Spring docs: Bean Scopes - Singleton scope</a></li>
```
Comment (nvhaver) [inline recipes/Java/Spring/Boot/descriptions/Code_quality__Do_not_add__Scope__singleton___on_a_Spring__rest_controller__service_or_repository.html:18]:
```suggestion
  <li><a href="https://docs.spring.io/spring-framework/docs/current/reference/html/core.html#beans-factory-scopes">Spring docs: Bean Scopes</a></li>
```

File: recipes/Java/Spring/Boot/descriptions/Spring_recommendation____Rest_Controller___Service___Repository_should_be_singletons.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+}</pre>
+  <br>
+  <b>After</b>
+<pre>@Controller
+public class UserController {
+}</pre>
+  <br>
+  <b>Resources</b>
+<ul>
+  <li><a href="https://docs.spring.io/spring-framework/docs/3.0.0.M3/reference/html/ch04s04.html">Spring docs: Bean Scopes - Singleton scope</a></li>
```
Comment (nvhaver) [inline recipes/Java/Spring/Boot/descriptions/Spring_recommendation____Rest_Controller___Service___Repository_should_be_singletons.html:18]:
```suggestion
  <li><a href="https://docs.spring.io/spring-framework/docs/current/reference/html/core.html#beans-factory-scopes">Spring docs: Bean Scopes</a></li>
```

File: recipes/Java/Spring/Boot/Controller_Repository_Service_should_be_singletons.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+              matches: Controller|RestController|Repository|Service
+            checkInheritance: true
+    type: Scope
+    without:
+      parameters:
+      - value:
+          value:
+            stringified: singleton
+availableFixes:
+- name: Remove the annotation
```
Comment (nvhaver) [inline recipes/Java/Spring/Boot/Controller_Repository_Service_should_be_singletons.yaml:28]:
```suggestion
- name: Remove the annotation (defaults to singleton scope)
```

**Pull request merged**